public/app/features/dashboard/components/LoadingBar/LoadingBar.tsx

```diff
@@ -0,0 +1,54 @@
+import { css, keyframes } from '@emotion/css';
+import React from 'react';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { useStyles2 } from '@grafana/ui/src/themes';
+
+/**
+ * @internal
+ */
+export interface LoadingBarProps {
+  width?: string;
+  height?: string;
+  ariaLabel?: string;
+}
+
+/**
+ * @internal
+ */
+export function LoadingBar({ width, height, ariaLabel = 'Loading bar' }: LoadingBarProps) {
+  const styles = useStyles2(getStyles(width, height));
+
+  return (
+    <div className={styles.container}>
+      <div aria-label={ariaLabel} className={styles.bar} />
+    </div>
+  );
+}
+
+const getStyles = (width?: string, height?: string) => (_: GrafanaTheme2) => {
+  const barWidth = width ?? '128px';
+  const loadingHeigth = height ?? '2px';
+
+  const loadingAnimation = keyframes({
+    '0%': {
+      transform: 'translateX(0)',
+    },
+    '100%': {
+      transform: `translateX(calc(100% - ${barWidth}))`,
+    },
+  });
+
+  return {
+    container: css({
+      width: '100%',
+      animation: `${loadingAnimation} 2s infinite linear`,
```

Comment:
Why the animation is applied on the container, not the bar itself?

Reply:
Because the container "knows" how wide their allowed space is, because it has `width: 100%`.
The bar even if it translates itself to 100% of its x-axis, it would not be connected to the container's width. We have to re-calculate the container's width too often to know how much the bar should move, which is expensive.

Reply:
Hey @dprokop 👋🏾 , we gave it a try with your approach, and unfortunately, it does not work as expected, it requires an `overflow: hidden` to the container (that is `PanelChrome` in our case), because the second bar `left:-100%` is visible, take a look at the video to see the problem. We are going with our approach :), but thanks for the suggestion. 

https://user-images.githubusercontent.com/239999/207058024-e3e83f94-4a5f-461b-840e-f1d9ae92c816.mp4